### PR TITLE
feat: add fox farming v4

### LINKED ISSUE #2121: FoxPage: Fox Farming v4
### Overview

Fox Page should support the new Fox Farming v4, following the `foxfarm` support:

https://github.com/shapeshift/foxfarm/pull/98

### References and additional details

https://fox.shapeshift.com/fox-farming

Reference web implementation: https://github.com/shapeshift/web/pull/1927

### Acceptance Criteria

- The Fox Page fetches and displays the Fox Farming V4 APY data under the "Farming" section
- The "ETH-FOX V4 Farm" title is displayed
- The "Get Started" link routes to https://fox.shapeshift.com/fox-farming/liquidity/0x470e8de2ebaef52014a47cb5e6af86884947f08c/staking/0x24fd7fb95dc742e23dc3829d3e656feeb5f67fa0
- The data is consistent to the one in the /fox-farming page

### Need By Date

_No response_

### Screenshots/Mockups

image

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This:
- adds the fox farming v4 APR-fetching logic
- adds the opportunity as a FOX farming opportunity in the Fox Page
- uses this opportunity as instead of the V2 opportunity one in the FoxFarm CTA
- abstracts the LP-APR-making logic into its own util to make it reusable 
- explicitly refers to opportunities as v2/v4

Note, this doesn't remove the V2 opportunity.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2121

## Risk

None

## Testing

- The Fox farming V4 contract opportunity should be shown in the Fox Page under farming opportunities with an APR matching the one from https://fox.shapeshift.com/fox-farming (small percentage discrepancies can happen between the two)
- The Fox Farming CTA in DeFi earn should display the V4 contract APR

## Screenshots (if applicable)

image
image

### CODE REVIEW

File: src/plugins/foxPage/utils.ts [hunk 1]
```diff
@@ -122,3 +122,13 @@ export const rewardRatePerToken = memoize(async (farmingRewardsContract: Contrac
     throw new Error(errorMsg)
   }
 })
+
+export const makeLpApr = (foxRewardRatePerToken: string, foxEquivalentPerLPToken: string) =>
```
Comment (elmutt) [inline src/plugins/foxPage/utils.ts:126]: maybe makeTotalLpApr is a better some as it is the combination of farming rate + LP rate ?

File: src/plugins/foxPage/hooks/useFarmingApr.ts [hunk 2]
```diff
@@ -8,69 +8,77 @@ import {
...
 
   useEffect(() => {
     if (
       !ethersProvider ||
       !Fetcher ||
       !blockNumber ||
-      !uniswapLPContract ||
-      !farmingRewardsContract
+      !uniV2LPContract ||
+      !farmingRewardsV2Contract
```
Comment (0xdef1cafe) [inline src/plugins/foxPage/hooks/useFarmingApr.ts:49]:
```suggestion
      !farmingRewardsV2Contract ||
      !farmingRewardsContractV4
```

**Comment by gomesalexandre** [comment]
> looks good. Curious what happened to v3? I see v2 and v4 but nothing about v3

Yeah that's just terminology, it was initially named V3 but renamed in https://github.com/shapeshift/web/pull/2063
tl;dr we mix up the UniV2 / UniV3 and the Fox Farm V3/V4 terminologies

File: src/plugins/foxPage/hooks/useFarmingApr.ts [hunk 3]
```diff
@@ -8,69 +8,77 @@ import {
...
+    () => new Contract(uniV2LiquidityContractAddress, IUniswapV2Pair.abi, ethersProvider),
+    [uniV2LiquidityContractAddress],
   )
-  const farmingRewardsContract = useMemo(
+
+  const farmingRewardsV2Contract = useMemo(
     () => new Contract(UNISWAP_V2_WETH_FOX_FARMING_REWARDS_ADDRESS, farmingAbi, ethersProvider),
     [],
   )
+  const farmingRewardsContractV4 = useMemo(
```
Comment (0xdef1cafe) [inline src/plugins/foxPage/hooks/useFarmingApr.ts:38]: should we name these consistently?

File: src/plugins/foxPage/hooks/useOtherOpportunities.ts [hunk 4]
```diff
@@ -34,14 +34,24 @@ export const useOtherOpportunities = (assetId: AssetId) => {
...
               icons: [
                 'https://assets.coincap.io/assets/icons/eth@2x.png',
                 'https://assets.coincap.io/assets/icons/fox@2x.png',
               ],
             },
+            {
+              title: 'ETH-FOX UNI V4 Farm',
+              isLoaded: isFarmingAprV4Loaded,
+              apy: isFarmingAprV4Loaded ? farmingAprV4 : null,
+              link: 'https://fox.shapeshift.com/fox-farming/liquidity/0x470e8de2ebaef52014a47cb5e6af86884947f08c/staking/0x24fd7fb95dc742e23dc3829d3e656feeb5f67fa0',
```
Comment (0xdef1cafe) [inline src/plugins/foxPage/hooks/useOtherOpportunities.ts:49]: should this also go to `/get-started`?

File: src/plugins/foxPage/utils.ts (hunk 1 above)
Comment (gomesalexandre) [reply src/plugins/foxPage/utils.ts]: Good idea! Always better to go explicit vs. broad naming, updated in  4afa9188a9f1dbc0fc9218ed6208e5e2965b6a4b

File: src/plugins/foxPage/hooks/useFarmingApr.ts (hunk 3 above)
Comment (gomesalexandre) [reply src/plugins/foxPage/hooks/useFarmingApr.ts]: Done in 92a6de37093ea2d5dd83068398ed74ee9168c551

File: src/plugins/foxPage/hooks/useOtherOpportunities.ts (hunk 4 above)
Comment (gomesalexandre) [reply src/plugins/foxPage/hooks/useOtherOpportunities.ts]: Nice catch ser! Updated in 92a6de37093ea2d5dd83068398ed74ee9168c551 and also updated the issue spec to match it!
